procedures/automation-android-setup.md

```diff
@@ -0,0 +1,200 @@
+                    * `service_account` set to "dummy"
+                    * `certificate` set to "dummy" 
+                    * `certificate_target_location` set to `${root}/$product.p12`
+                2. In `$google_play_accounts_config_content`, add a new dictionary at `$product` with properties that point `service_account` and `certificate` to the values in `$google_play_config`
+                3. In `$jarsigner_certificate_aliases_content`, add a mapping from `$product` to the same string: `"$product"`
+            3. In `$pushapk_scriptworker_env` for `mobile-prod`
+                1. In `$google_play_config`, add a new dictionary at `$product` with properties:
+                    * `service_account` set to `$_google_play_accounts['$product']['service_account']` (so it fetches from `hiera`)
+                    * `certificate` set to `$_google_play_accounts['$product']['certificate']` (so it fetches from `hiera`)
+                    * `certificate_target_location` set to `${root}/$product.p12`
+                2. In `$google_play_accounts_config_content`, add a new dictionary at `$product` with properties that point `service_account` and `certificate` to the values in `$google_play_config`
+                3. In `$jarsigner_certificate_aliases_content`, add a mapping from `$product` to the same string: `"$product"`
+            4. In `$jarsigner_all_certificates`
+                1. Add two entries:
+                    * `'$product-dep' => "${root}/$product_dep.cer"`
+                    * `'$product-release' => "${root}/$product_release.cer"`
+        3. In `manifests/init.pp`
+            1. For both `mobile-dep` and `mobile-prod`, add an entry so that at `certificate_target_location`, the `content` is set to `certificate` for your product
+        4. In `manifests/jarsigner_init.pp`, for both `mobile-dep` and `mobile-prod`:
+            1. For convenience, set a variable at the top of the section that points to the relevant certificate location (either `dep` or `prod` depending on the section you're currently on)
+            2. Add an entry to the `file` block so that, at the certificate location, the source of the correct `pem` file is copied in
+            3. Add an entry to the `java_ks` block for your product, setting `certificate` to your certificate location
+    5. In `files/requirements.txt`
+        1. From step 9, update the version of `mozapkpublisher` and `pushapkscript`
+11. Commit and push your `build-puppet` changes, make a PR
+12. Test your automation with staging releases
+    1. Confirm with the team that nobody is using the dep signing or pushapk workers
+    2. SSH into the puppet master
+    3. Go to `/etc/puppet/environments/$username`
+    4. Temporarily move files out of the directory (like `environment.conf`)
+    5. `git clone https://github.com/$username/build-puppet .` to clone `build-puppet` to the directory
```

Comment:
Tip: You may want to add the upstream repo too. This helps in getting the latest changes when you want to work on a new bug.

Reply:
Hmm, I didn't realize that this section was on the puppet master. You have `upstream` configured there? I keep my local `build-puppet` up to date and strictly pull on my repo on `build-puppet` so I don't have conflicts.

Both approaches are valid, but since we aren't consistent here, I'll leave it up to the reader :)